8.11 release notes (#4095)
elastic/security-docs: docs/release-notes/8.11.asciidoc

[[release-notes-header-8.11.0]]
== 8.11

[discrete]
[[release-notes-8.11.0]]
=== 8.11.0

[discrete]
[[breaking-changes-8.11.0]]
==== Breaking changes
* Ends support for the `filterQuery` field of the `getLiveQueryResults` and `findLiveQuery` APIs, and replaces it with the KQL field `kuery`. Requests to those APIs that used the `filterQuery` field should replace it with `kuery` ({pull}161806[#161806]).

> * In 8.11, rule APIs will only support `investigation_fields` as { field_names: string[] }. Note that, if you've added this field to your rules in 8.10, you don't need to do anything when you import your rules. 

Reviewer: I think this example should be monospace too -- kinda awkward to read though. 🤷 Also, does the last part apply _only_ to 8.10, or is it something like "8.10 or earlier"?
```suggestion
* In 8.11, rule APIs will only support `investigation_fields` as `{ field_names: string[] }`. If you've added this field to your rules in 8.10, you don't need to do anything when you import your rules. 
```

[discrete]
[[deprecations-8.11.0]]
==== Deprecations
* Deprecates the `doc_root.vulnerability.package` and replaces it with the `doc_root.package` ECS package ({pull}164651[#164651]).

[discrete]
[[features-8.11.0]]
==== New features

> * Upgrades {elastic-defend} for Windows to capture a new event type: ETW Threat Intelligence (ETW-TI). Renames the Windows events policy `Credential access` category to `API` in the UI (but not in the `.yaml`, maintaining backwards compatibility). Adds two new advanced options: `windows.advanced.events.api_disabled` and

Reviewer: To avoid sounding like we have a product called "Elastic Defend for Windows" (pretty sure it's just "Elastic Defend"):
```suggestion
* Upgrades {elastic-defend} to capture a new Windows event type: ETW Threat Intelligence (ETW-TI). Renames the Windows events policy `Credential access` category to `API` in the UI (but not in the `.yaml`, maintaining backwards compatibility). Adds two new advanced options: `windows.advanced.events.api_disabled` and
```

`windows.advanced.events.api_verbose` ({pull}167549[#167549]).
* Adds the `Same family` category and tab to the Data Quality dashboard. Fields with mappings in the same family have the same search behavior as the type specified by ECS, but may have different space usage or performance characteristics ({pull}167480[#167480]).
* Updates the exceptions flyout's `match_any` operator to accept duplicate values that differ in case. Previously, values of `foo` and `FOO` were incorrectly considered duplicates ({pull}167208[#167208]).
* Enables the Elastic AI Assistant to answer questions about Elasticsearch Query Language (ES|QL) by allowing it to query, via ELSER, an ES|QL knowledge base. Refer to <<security-assistant, Elastic AI Assistant>> to enable the knowledge base ({pull}167097[#167097]).

> * Enables ES|QL in Timeline (technical preview) ({pull}166764[#166764]).

Reviewer 2: A lot of the discover functionality was disabled. (i.e. you can't switch dataviews anymore since your locked to ESQL, and the top bar actions such as `save`, `share`, `inspect`, etc... aren't present here as well so I wouldn't highlight it

* Adds the new ES|QL rule type (technical preview) ({pull}165450[#165450]).
* Implements Kafka output for {elastic-endpoint} (https://github.com/elastic/endpoint-dev/issues/13192[#13192]).

> * Allows {elastic-endpoint} to support MacOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).

Reviewer 3: @roxana-gheorghe let me know if you agree - I feel like we maybe should get a little more specific here (seems to imply we're only now supporting Mac?)

> * Allows {elastic-endpoint} to support MacOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).

Author: ```suggestion
* Introduces full support for {elastic-endpoint} on MacOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).
```

> * Introduces full support for {elastic-endpoint} on MacOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).

Reviewer: ```suggestion
* Introduces full support for {elastic-endpoint} on macOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).
```

[discrete]
[[enhancements-8.11.0]]
==== Enhancements
* Adds a new Generative AI connector, Amazon Bedrock, for use with Elastic AI Assistant ({pull}166662[#166662]).

> * Adds fields to the Cases webhook: `id`, `severity`, and `status` ({pull}166295[#166295]).

Reviewer 4: ```suggestion
* Adds the `id`, `severity`, and `status` fields to the Webhook - Case Management connector ({pull}166295[#166295]).
```

> * Updates the order of items on {kib}'s left-side navigation menu to match the order in {elastic-sec}'s' left-side navigation menu ({pull}164268[#164268]).

Reviewer 4: a sp00ky extra apostrophe sighting!

```suggestion
* Updates the order of items on {kib}'s left-side navigation menu to match the order in {elastic-sec}'s left-side navigation menu ({pull}164268[#164268]).
```

* Updates the Endpoint policy UI (**Manage -> Policies**) to include a `Protection updates` tab, a new column called `Deployed version`, and a banner that highlights outdated policies ({pull}165256[#165256], {pull}162719[#162719]).
* Adds tooltips to overview section titles in the alert details flyout ({pull}166737[#166737]).

> 

Reviewer 4: ```suggestion
The `.lists` and `.items` indices were moved to data streams ({pull}162508[#162508]).
```

> * Adds tooltips to link titles on the expandable alerts flyout ({pull}166737[#166737]).

Reviewer 4: ```suggestion
* Adds tooltips to overview section titles in the alert details flyout ({pull}166737[#166737]).
```

> The `.lists` and `.items` indices were moved to data streams ({pull}162508[#162508]).

Author: ```suggestion
* Updates the `.lists` and `.items` indices to data streams ({pull}162508[#162508]).
* Renames the Generative AI connector to OpenAI, since Generative AI is now a category of connectors that include OpenAI and Amazon Bedrock ({pull}167677[#167677]).
```

[discrete]
[[bug-fixes-8.11.0]]
==== Bug fixes
* Updates the Entity Risk Score error message to list the necessary permissions ({pull}169216[#169216]).

> * Renames the Generative AI connector to OpenAI, since Generative AI is now a category of connectors that include OpenAI and Amazon Bedrock ({pull}167677[#167677]).

Reviewer 4: Is this a bug fix? Seems more like an enhancement to me.

> * Renames the Generative AI connector to OpenAI, since Generative AI is now a category of connectors that include OpenAI and Amazon Bedrock ({pull}167677[#167677]).

Author: ```suggestion
```

* Displays more descriptive errors for Generative AI connectors ({pull}167674[#167674]).
* Adds metrics to some rule execution warning messages ({pull}167551[#167551]).
* Fixes a bug that could cause the exceptions flyout to reload unnecessarily in response to rule updates ({pull}166914[#166914]).
* Fixes a bug that could cause EQL shell alerts to not include certain common fields ({pull}166751[#166751]).

> * Makes the prevalence details datepicker appear in full width on the expandable alerts flyout ({pull}166714[#166714]).

Reviewer 4: ```suggestion
* Sets the date time picker to full width in the expanded Prevalence view within the alert details flyout ({pull}166714[#166714]).
```

> * Fixes a bug that could prevent the "Install Cloud Native Vulnerability Management" button on the empty state of the Findings page from working ({pull}166335[#166335]).

Reviewer 4: ```suggestion
* Fixes a bug that could prevent the **Install Cloud Native Vulnerability Management** button on the empty state of the Findings page from working ({pull}166335[#166335]).
```

* Fixes a bug that could cause an error when you edited a rule's filter ({pull}165262[#165262]).

> * Fixes a bug that caused the rules table to auto-refresh when auto-refresh was disabled ({pull}165250[#165250]).

Reviewer 4: ```suggestion
* Fixes a bug that caused the Rules table to auto-refresh when auto-refresh was disabled ({pull}165250[#165250]).
```

[[release-notes-header-8.11.0]]
== 8.11

[discrete]
[[release-notes-8.11.0]]
=== 8.11.0

[discrete]
[[known-issue-8.11.0]]
==== Known issues

> * Duplicate rules are displayed in MITRE ATT&CK® technique cells ({issue}167929[#167929]).
> * The incorrect rule count is shown in MITRE ATT&CK® technique cells ({issue}167930[#167930]).
> * The incorrect MITRE ATT&CK® sub-technique is applied after saving a rule ({issue}170347[#170347]).

Reviewer 5: Consider consolidating these into succinct bullets a little:
```suggestion
For MITRE ATT&CK® technique cells:
* Duplicate rules are displayed ({issue}167929[#167929]).
* The incorrect rule count is shown ({issue}167930[#167930]).
* The incorrect MITRE ATT&CK® sub-technique is applied after saving a rule ({issue}170347[#170347]).
```

Reviewer 4: I think the last item is tied to the rule creation flow, but the first two can be grouped into a list. 

```suggestion
* For MITRE ATT&CK® technique cells:
** Duplicate rules are displayed ({issue}167929[#167929]).
** The incorrect rule count is shown ({issue}167930[#167930]).
* The incorrect MITRE ATT&CK® sub-technique is applied after saving a rule ({issue}170347[#170347]).
```

* In 8.11, rule APIs will only support `investigation_fields` as `{ field_names: string[] }`. If you've added this field to your rules in 8.10, you don't need to do anything when you import your rules. 

[discrete]
[[deprecations-8.11.0]]
==== Deprecations
* Deprecates the `doc_root.vulnerability.package` and replaces it with the `doc_root.package` ECS package ({pull}164651[#164651]).

[discrete]
[[features-8.11.0]]
==== New features
* Upgrades {elastic-defend} to capture a new Windows event type: ETW Threat Intelligence (ETW-TI). Renames the Windows events policy `Credential access` category to `API` in the UI (but not in the `.yaml`, maintaining backwards compatibility). Adds two new advanced options: `windows.advanced.events.api_disabled` and
`windows.advanced.events.api_verbose` ({pull}167549[#167549]).
* Adds the `Same family` category and tab to the Data Quality dashboard. Fields with mappings in the same family have the same search behavior as the type specified by ECS, but may have different space usage or performance characteristics ({pull}167480[#167480]).

> * Updates the exceptions flyout's `match_any` operator to accept duplicate values that differ in case ({pull}167208[#167208]).

Reviewer 6: 👍🏽

* Adds the new ES|QL rule type (technical preview) ({pull}165450[#165450]).
* Updates the Endpoint policy UI (**Manage -> Policies**) to include a `Protection updates` tab, a new column called `Deployed version`, and a banner that highlights outdated policies ({pull}165256[#165256], {pull}162719[#162719]).
* Implements Kafka output for {elastic-endpoint} (https://github.com/elastic/endpoint-dev/issues/13192[#13192]).
* Introduces full support for {elastic-endpoint} on macOS (https://github.com/elastic/endpoint-dev/issues/13058[#13058]).

[discrete]
[[enhancements-8.11.0]]
==== Enhancements
* Adds a new Generative AI connector, Amazon Bedrock, for use with Elastic AI Assistant ({pull}166662[#166662]).
* Adds the `id`, `severity`, and `status` fields to the Webhook - Case Management connector ({pull}166295[#166295]).
* Updates the order of items on {kib}'s left-side navigation menu to match the order in {elastic-sec}'s left-side navigation menu ({pull}164268[#164268]).
* Adds tooltips to overview section titles in the alert details flyout ({pull}166737[#166737]).
* Updates the `.lists` and `.items` indices to data streams ({pull}162508[#162508]).

> * Renames the Generative AI connector to OpenAI, since Generative AI is now a category of connectors that include OpenAI and Amazon Bedrock ({pull}167677[#167677]).

Reviewer 5: I would put this after line 43, with the first AI bullet.

Reviewer 4: Fixed in [b46277b](https://github.com/elastic/security-docs/pull/4095/commits/b46277b1e86de3c9e4204e2555b941ba6fa4a74c).

[discrete]
[[bug-fixes-8.11.0]]
==== Bug fixes
* Updates the Entity Risk Score error message to list the necessary permissions ({pull}169216[#169216]).
* Displays more descriptive errors for Generative AI connectors ({pull}167674[#167674]).
* Adds metrics to some rule execution warning messages ({pull}167551[#167551]).
* Fixes a bug that could cause the exceptions flyout to reload unnecessarily in response to rule updates ({pull}166914[#166914]).
* Fixes a bug that could cause EQL shell alerts to not include certain common fields ({pull}166751[#166751]).

> * Sets the date time picker to full width in the expanded Prevalence view within the alert details flyout ({pull}166714[#166714]).

Reviewer 5: ```suggestion
* Sets the date and time picker to full width in the expanded Prevalence view within the alert details flyout ({pull}166714[#166714]).
```

[[release-notes-header-8.11.0]]
== 8.11

[discrete]
[[release-notes-8.11.0]]
=== 8.11.0

[discrete]
[[known-issue-8.11.0]]
==== Known issues
* For MITRE ATT&CK® technique cells:
** Duplicate rules are displayed ({issue}167929[#167929]).

> ** The incorrect rule count is shown ({issue}167930[#167930]).

Reviewer 7: The incorrect rule count is technically in the tactic cell, not the technique cell. All these MITRE issues can probably be bullet points on the same level.
